sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,413 @@ def reduce_inequalities(inequalities, symbols=[]):
+
+    return A, B, C
+
+# Maximize Cx + D constrained with Ax <= B and x >= 0
+# Minimize y^{T}B constrained with y^{T}A >= C^{T} and y >= 0
+def linear_programming(constraints, objective):
+    """
+    A function to maximize a linear objective function subject to linear
+    constraints with the simplex method.
+
+    Explanation
+    ===========
+
+    When x is a column vector of variables, y is a column vector of dual variables,
+    and when the objective is either:
+
+    *) Maximizing Cx constrained to Ax <= B and x >= 0.
+    *) Minimizing y^{T}B constrained to y^{T}A >= C^{T} and y >= 0.
+
+    The method returns a triplet of solutions optimum, argmax and argmax_dual where
+    optimum is the optimum solution, and argmax/ argmax_dual give the
+    values for x and y, respectively.
+
+    Parameters
+    ==========
+
+    constraints : list of linear inequalities
+        The lhs of these inequalities are equivalent to the matrix A and the rhs
+        are equivalent to the column vector B.
+
+    objective : a linear function to maximize
+        The objective function is represented by the row vector C.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import linear_programming
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual = linear_programming([r1,r2,r3], x+y+5*z)
+    >>> optimum
+    11/3
+    >>> argmax
+    [0, 1/3, 2/3]
+    >>> argmax_dual
+    [0, 2/3, 1]
+
+    The method also works with non-integer coefficients (like sqrt(2)).
+
+    >>> from sympy import sqrt
+    >>> r1 = y + sqrt(2)*z <= sqrt(3)
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmax_dual = linear_programming([r1,r2,r3], x+y+sqrt(5)*z)
+    >>> optimum
+    3
+    >>> argmax
+    [2, 1, 0]
+    >>> argmax_dual
+    [0, 1, 1]
+
+    See Also
+    ========
+
+    solve_univariate_inequality
+    find_feasible
+    """
+    from sympy.matrices.dense import Matrix
+    A, B, C = _to_standard_form(constraints, objective)
+    D = ImmutableMatrix([0])
+    M = Matrix([[A, B], [-C, D]])
+    r_orig = ['x_{}'.format(j) for j in range(M.cols - 1)]
+    s_orig = ['y_{}'.format(i) for i in range(M.rows - 1)]
+
+    r = r_orig.copy()
+    s = s_orig.copy()
+
+    M, r, s = _simplex(M, r, s)
+
+    argmax = []
+    argmin_dual = []
+
+    for x in r_orig:
+        for i, xx in enumerate(s):
+            if x == xx:
+                argmax.append(M[i, -1])
+                break
+        else:
+            argmax.append(S.Zero)
+
+    for x in s_orig:
+        for i, xx in enumerate(r):
+            if x == xx:
+                argmin_dual.append(M[-1, i])
+                break
+        else:
+            argmin_dual.append(S.Zero)
+    return M[-1, -1], argmax, argmin_dual
+
+
+def find_feasible(constraints):
```

Comment:
Most of this function is identical to linear_programming. I'll probably add some helper functions to reduce duplicate code. 

Also, I'm not sure if this belongs more to sympy/logic rather than sympy/solvers.